rxandroidble/src/main/java/com/polidea/rxandroidble/internal/operations/CharacteristicLongWriteOperation.java

```diff
@@ -62,12 +67,14 @@
         this.bluetoothGattCharacteristic = bluetoothGattCharacteristic;
         this.batchSizeProvider = batchSizeProvider;
         this.writeOperationAckStrategy = writeOperationAckStrategy;
+        this.writeOperationRetryStrategy = writeOperationRetryStrategy;
         this.bytesToWrite = bytesToWrite;
+        this.retryCounter = 0;
```

Comment:
I am not a big fan of variables that are changed as a side effect. This usually makes room for a mistake and makes the code harder to trace.

Reply:
You mean we should not have a separated counter in this Operation class at all?

Reply:
I believe that Darek meant putting a reset in a `doOnNext`. AFAIK you can safely reset the counter before the rx subscription (around line 86.)